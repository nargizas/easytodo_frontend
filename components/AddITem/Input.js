import { View, Text, TextInput, StyleSheet } from "react-native";
import { GlobalStyles } from "../../constants/styles";

function Input({ label, textInputConfig }) {
    return (<View style={styles.inputContainer}>
        <Text style={styles.label}>{label}</Text>
        <TextInput {...textInputConfig} style={styles.input} />
    </View>
    );
}

export default Input;

const styles = StyleSheet.create({
    input: {
        borderRadius: 8,
        paddingHorizontal: 16,
        paddingVertical: 8,
        backgroundColor: "#f5f5f5",
        fontSize: 18,
        elevation: 2
    },
    label: {
        fontSize: 18,
        color: GlobalStyles.colors.grey,
        marginBottom: 4,
    },
    inputContainer: {
        // flexDirection: 'column',
        marginHorizontal: 16,
        marginVertical: 8,
    }
});