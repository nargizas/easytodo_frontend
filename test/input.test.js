import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../components/AddITem/Input';

it('renders correctly', () => {
  const tree = renderer.create(<Input label="abca" />).toJSON();
  expect(tree).toMatchSnapshot();
});
