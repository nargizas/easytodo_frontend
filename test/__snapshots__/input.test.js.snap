// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "marginHorizontal": 16,
      "marginVertical": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#f5f5f5",
        "fontSize": 18,
        "marginBottom": 4,
      }
    }
  >
    abca
  </Text>
  <TextInput
    style={
      {
        "backgroundColor": "#f5f5f5",
        "borderRadius": 8,
        "elevation": 2,
        "fontSize": 18,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  />
</View>
`;
